tests/Fixtures/AlgoliaEngineTestModelCustomKey.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Tests\Fixtures;
+
+class AlgoliaEngineTestModelCustomKey extends TestModel
+{
+    public function getSearchableKey()
+    {
+        return "custom-{$this->getKey()}";
+    }
+
+    public function getReverseSearchableKey($key = '')
+    {
+        return str_replace('custom-1', '', $key);
```

Comment:
It is pretty much the same test as `test_map_correctly_maps_results_to_models` but with a custom ObjectID. I am not directly testing the `getSearchableKey()` & `getReverseSearchableKey()` methods, as these methods are on the Fixture model class.

Just as the `test_map_correctly_maps_results_to_models` method, the model is mocked and so is the `getReverseSearchableKey` method, which explains why it would always return true even if you change the method on the Fixture. If you change the return of the mocked method though, it will fail:

`$model->shouldReceive('getReverseSearchableKey')->andReturn(2);`

Maybe `test_map_correctly_maps_results_to_models_with_custom_keys` is not needed since we already have `test_map_correctly_maps_results_to_models` as they are pretty much the same.

Also, the map method on the engine is fetching the list of ObjectID from the hits, which could be custom and not match the ID used in the database when the engine runs a query in the DB to fetch all the models within an array of ObjectIDs. 

So either the map method in the Engine class should list the ID instead of ObjectIDs, but the ID would be required in the searchable fields. Currently, the ID field is required when indexing anyway.

Or we need a method on the model to transform an ObjectID into the ID used in the DB.